pkg/tfbridge/x/tokens.go

```diff
@@ -563,13 +576,17 @@ func aliasOrRenameResource(p *b.ProviderInfo, tfToken string, hist *tokenHistory
 	}
 	if !alreadyPresent {
 		hist.Past = append(hist.Past, alias[tokens.Type]{
-			Name:      hist.Current,
-			InCodegen: true,
+			Name:         hist.Current,
+			InCodegen:    true,
+			MajorVersion: currentVersion,
 		})
 	}
 	for _, a := range hist.Past {
 		legacy := a.Name
-		if a.InCodegen {
+		// A version set as "-1" means the provider did not have a version set
+		// so we refrain from removing aliases
+		fmt.Println(legacy, a.MajorVersion, currentVersion)
```

Comment:
We never remove old aliases from the history, which means that aliases multiple major versions back might exist. I think we can conservatively assume that the most recent major version referenced in the history is the most recent major version. This way unsetting the version won't introduce multiple generations of aliases into a build.

I appreciate the comment explaining what's going on!

Reply:
This makes more sense. In our tests previously, we didn't have a version set on the mock providers since it isn't a required attribute, so I wasn't sure if we needed to check for that.
I think it'd be better to default to the most recent version referenced if one doesn't exist.